prelude/prelude.purs

```diff
@@ -170,9 +228,30 @@ module Prelude
 
   infixl 1 >>=
 
+  -- | A Bind is an Apply with a bind operation which sequentially composes actions.
+  -- |
+  -- | A Bind should obey the following rule.
+  -- |
+  -- | Associative (as Bind)
+  -- |     forall f g x. (x >>= f) >>= g = x >>= (\k => f k >>= g)
+  -- |
   class (Apply m) <= Bind m where
     (>>=) :: forall a b. m a -> (a -> m b) -> m b
 
+  -- | Monad is a class which can be intuitively thought of as an abstract datatype of actions or
+  -- | more formally though of as a monoid in the category of endofunctors.
+  -- |
+  -- | A Monad should obey the following rules.
+  -- |
+  -- | Associative (as Bind)
```

Comment:
I wonder if it's worth writing these out in `do` notation as well. I always found them easier to understand that way.